File: client/src/app/entities/vehicle.ts
import {User} from './user';
import {DataPage} from '../shared/data-page';
import {Province} from './province';
import {Vehiclemodel} from './vehiclemodel';
import {Vehiclecondition} from './vehiclecondition';
import {Fueltype} from './fueltype';
import {Vehicletype} from './vehicletype';
import {Country} from './country';
import {Brand} from './brand';

export class Vehicle {
  id: number;
  code: string;
  chassisnumber: string;
  enginenumber: string;
  yearofmanufacture: string;
  dateofregistration: string;
  color: string;
  originalcolor: string;
  description: string;
  enginecapacity: string;
  fueltype: Fueltype;
  vehiclemodel: Vehiclemodel;
  province: Province;
  registrationno: string;
  tocreation: string;
  vehiclecondition: Vehiclecondition;
  creator: User;
  country:Country;
  vehicletype: Vehicletype;
  brand: Brand;
  seatingcapacity: string;

  constructor(id: number = null) {
    this.id = id;
  }
}
export class VehicleDataPage extends DataPage{
    content: Vehicle[];
}

File: server/src/main/java/bit/project/server/entity/Vehicle.java
package bit.project.server.entity;


import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;


@Data
@Entity
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Vehicle {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private String chassisnumber;

    private String enginenumber;

    private LocalDate yearofmanufacture;

    private LocalDate dateofregistration;

    private String color;

    private String originalcolor;

    @Lob
    private String description;

    private String enginecapacity;

    @ManyToOne
    private Fueltype fueltype;

    @ManyToOne
    private Vehiclemodel vehiclemodel;

    @ManyToOne
    private Province province;

    private String registrationno;

    private LocalDateTime tocreation;

    @ManyToOne
    private Vehiclecondition vehiclecondition;

    @ManyToOne
    private Country country;


    @ManyToOne
    private Vehicletype vehicletype;

    @ManyToOne
    private Brand brand;

    private String seatingcapacity;


    @JsonIgnore
    @OneToMany(mappedBy = "vehicle")
    private List<Offerrequest> vehicleList;

    @JsonIgnore
    @OneToMany(mappedBy = "vehicle")
    private List<Valuation> valuationList;

    @ManyToOne
    @JsonIgnoreProperties({"creator","tocreation","roleList"})
    private User creator;


    public Vehicle(Integer id) {
        this.id = id;
    }

    public Vehicle(Integer id,String code, String registrationno, String chassisnumber, String enginenumber){
        this.code = code;
        this.registrationno = registrationno;
        this.chassisnumber = chassisnumber;
        this.enginenumber = enginenumber;
        this.id = id;

    }

}
